### PR TITLE
Validate null character - Closes #1108

### LINKED ISSUE #1108: Disallow null character on transfer and dapp transaction
### Expected behavior
Expect Dapp registration transaction to be rejected if using unicode null character (`\u0000`) in `tag` or `description` attributes.

### Actual behavior
There is no check regards to the null character in Dapp registration

### Steps to reproduce
Post a Dapp registration transaction with `\u0000` in the `description` or `tag` attribute

Related to https://github.com/LiskHQ/lisk/issues/2886

### Which version(s) does this affect? (Environment, OS, etc...)
2.1-alpha.0

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
By protocol, Transfer Transaction and Dapp transaction should not allow null character in string field.

### How did I fix it?
Add the same check done as https://github.com/LiskHQ/lisk/commit/5c7e2944383bf369f1343fba62c71e84de6f9664.
Also, added test to check schema validation works.

### How to test it?
`npm t`

### Review checklist

* The PR resolves #1108 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Review by lsilvs — COMMENTED** [review]
Should `noNullByte` check be also included for delegate username?
In Core we are doing so. It's not really required because it doesn't allow `\` character but we included for consistency.
Other than that, looks good :)

File: packages/lisk-transactions/test/utils/validation/validator.ts [hunk 1]
```diff
@@ -592,4 +592,47 @@ describe('validator', () => {
 			).to.be.false;
 		});
 	});
+
+	describe('noNullBytes', () => {
+		let validate: ValidateFunction;
+		beforeEach(() => {
```
Comment (mitsuaki-u) [inline packages/lisk-transactions/test/utils/validation/validator.ts:598]: prefer `async` and no promise for consistency. same for below

**Review by mitsuaki-u — APPROVED** [review]

**Review by lsilvs — APPROVED** [review]

**Comment by webmaster128** [comment]
The term _noNullByte_ is highly misleading as you are checking JavaScript strings, which is a list of unicode characters. There is no _byte_ in a string. A string can be converted to a list of bytes by encoding it but a string itself is independent of any binary representation.

So this should be called _noNullCharacter_.

File: packages/lisk-transactions/test/utils/validation/validation.ts [hunk 2]
```diff
@@ -369,4 +370,47 @@ describe('validation', () => {
 			return expect(isValidInteger(-6)).to.be.true;
 		});
 	});
+
+	describe('#isNullByteIncluded', () => {
+		const validStrings = [
+			'lorem ipsum',
+			'lorem\u0001 ipsum',
+			'loremU00000001 ipsum',
```
Comment (webmaster128) [inline packages/lisk-transactions/test/utils/validation/validation.ts:378]: 'U00000000' should also be valid but it is not (see https://github.com/LiskHQ/lisk/commit/5c7e2944383bf369f1343fba62c71e84de6f9664)

**Pull request merged**